### PR TITLE
Snap/with-plugins.md

### PR DESCRIPTION
**Pull request by rashmikampurath**
@rizdaprasetya : This file is ready for your review.

### CODE REVIEW

File: en/snap/with-plugins.md [hunk 1]
```diff
@@ -56,269 +57,309 @@ Here is the list of Content Management System plugins and extensions that are su
...
+2. Select the **Environment** from drop-down list.
+3. On the Home page, go to **SETTINGS > CONFIGURATION**.
+   *Configuration* page is displayed.
+4. Enter **Payment Notification URL**.
+5. Enter **Finish Redirect URL**.
+6. Enter **Error Redirect URL**.
+7. Enter **Unfinish Redirect URL**.
+8. Click **Update**.
+   A confirmation message is displayed.
+   ![Snap API](./../../asset/image/snap-api-notification-configuration-1.png)
```
Comment (rizdaprasetya) [inline en/snap/with-plugins.md:135]: I saw lots of broken images throughout this page:

> Screen Shot 2020-11-16 at 16 32 19

The file `snap-api-notification-configuration-1.png` is missing.
What is this image about? screenshot of MAP notification url config pages? I think the instruction is clear enough, we don't need the image. Also because currently the pages is already have many images.

File: en/snap/with-plugins.md [hunk 2]
```diff
@@ -1,32 +1,33 @@
-# Integrate Midtrans Snap to Ecommerce Content Management System
+# Integrate Midtrans Snap to E-Commerce Content Management System
 <hr>
-Content Management System (CMS) allow you to easily have website / web store without building (programming) from scratch. Most of the time CMS don't require you to know too much about technical knowledge, by just installing the CMS and then configure or customize it to suit your desire. It means most of the time you just need to care about managing the content (Content Management). Example of CMS are: Wordpress, Magento 2, Prestashop, WHMCS, etc.
 
-### Preparation
+Content Management System (CMS) allows you to easily have a website/web store without building from scratch. CMS does not require programming knowledge.You just need to install the CMS and customize according your requirement. You only require to manage the content (Content Management). Some of the examples of CMS are WordPress, Magento 2, PrestaShop, WHMCS, and so on.
```
Comment (rizdaprasetya) [inline en/snap/with-plugins.md:4]: > knowledge.You

Missing space between `.` and `You`

File: en/snap/with-plugins.md [hunk 3]
```diff
@@ -56,269 +57,309 @@ Here is the list of Content Management System plugins and extensions that are su
...
+| 3 | Unfinish Redirect URL | [your-site-url]/?wc-api=WC_Gateway_Midtrans |
+| 4 | Error Redirect URL | [your-site-url]/?wc-api=WC_Gateway_Midtrans |
 
-> **Note:**
->
-> The `your-site-url` is where you install your Wordpress, it can be the domain root directory (e.g: `https://myshop.com` or `https://shop.myshop.com`) or within a sub directory (e.g: `https://myshop.com/wordpress/`)
->
-> Please make sure to input **http://** or **https://** when filling Notification URL and Redirect URL, according to your web-server configuration. 
-> 
-> If you are not sure, try opening your web URL in a browser, and check the URL is **http** or **https** on the address bar.
```
Comment (rizdaprasetya) [inline en/snap/with-plugins.md:122]: This sentence of tips got removed? this is important, please re-add.

File: en/snap/with-plugins.md [hunk 4]
```diff
@@ -350,68 +395,90 @@ Customer receives email notification| Receive notification|  Check MAP menu sett
...
-* MySQL version 5.0 or greater
-* Midtrans plugin for PrestaShop [ [Github](https://github.com/veritrans/SNAP-Prestashop) | [Zip](https://github.com/veritrans/SNAP-Prestashop/archive/master.zip) ].
 
-#### Installation Process:
+#### Requirements
+Some of the requirements to continue with the integration process, are listed below.
+* PrestaShop 1.6 and 1.7 or later
+* PHP version 5.4 or later
+* MySQL version 5.0 or later
+* Midtrans plugin for PrestaShop [[Github](https://github.com/veritrans/SNAP-Prestashop)|[Zip](https://github.com/veritrans/SNAP-Prestashop/archive/master.zip)].
```
Comment (rizdaprasetya) [inline en/snap/with-plugins.md:408]: > [[Github](https://github.com/veritrans/SNAP-Prestashop)|[Zip](https://github.com/veritrans/SNAP-Prestashop/archive/master.zip)]

Please do not remove the spacing from 
> [Github](https://github.com/veritrans/SNAP-Prestashop) | [Zip](https://github.com/veritrans/SNAP-Prestashop/archive/master.zip)

The spacing helps reader to know that there are 2 separate/different link, without spacing it looks like 1 same link.
This feedback applies to all the other parts in this page.

File: en/snap/with-plugins.md [hunk 5]
```diff
@@ -56,269 +57,309 @@ Here is the list of Content Management System plugins and extensions that are su
...
+
+#### Configuring Midtrans WooCommerce Plugin
+To configure WooCommerce plugin, follow the steps given below.
+1. Go to **WooCommerce > Settings > Payments > Midtrans** menu.
+2. Enter **Title**. This text appears on the button displayed to the customer.
+3. Select **Environment** from dropdown. `Sandbox` for testing transaction and `Production` for real transaction.
+4. Enter **Merchant ID**.
+5. Enter **Client Key**.
+6. Enter **Server key**.
+   For more details, refer to [Preparation](/en/snap/preparation.md).
```
Comment (rizdaprasetya) [inline en/snap/with-plugins.md:116]: Why did you breakdown the sub-step of 1 into its own numbered lists? it make the step numbers unnecessary look longer. Also it is a group of configurations step within a menu, it make sense to leave it as sub-step. 

Happen to other parts of this page too.

File: en/snap/with-plugins.md [hunk 6]
```diff
@@ -420,313 +487,436 @@ Customer receives email notification| Receive notification|  Check MAP menu sett
...
+* PHP version 5.4 or later
+* MySQL version 5.0 or later
+* Midtrans plugin for OpenCart
+
+#### Installation of OpenCart plugin
+To install OpenCart plugin, follow the steps given below.
+1. Download midtrans OpenCart plugin and extract the zip file.
+    * [OpenCart v3.0](https://github.com/Midtrans/Midtrans-Opencart3)
+    * [OpenCart v2.3](https://github.com/Midtrans/SNAP-Opencart-2.3/)
+    * [OpenCart v2.0, v2.1, v2.2](https://github.com/veritrans/SNAP-Opencart)
```
Comment (rizdaprasetya) [inline en/snap/with-plugins.md:507]: We want AptDocs to help with inconsistencies within our docs, for example, this part is inconsistent because it does not have ZIP download url, you should please add those to make it consistent with other CMS instructions.

e.g: ZIP download url is `https://github.com/Midtrans/Midtrans-Opencart3/archive/master.zip` for `https://github.com/Midtrans/Midtrans-Opencart3`

File: en/snap/with-plugins.md [hunk 7]
```diff
@@ -420,313 +487,436 @@ Customer receives email notification| Receive notification|  Check MAP menu sett
...
-   1. Download the modules from the link above.
-   2. Extract **Whmcs-master.zip** file you have previously downloaded.
-   3. Upload & merged module folder that you have extracted into your WHMCS directory. **Installation & Configuration**
-   4. Access your WHMCS admin page.
-   5. Go to `Setup` **->** `Payments` **->** `Payment Gateways` menu.
-   6. Click **Midtrans** payment method, then you will be redirected to configuration page.
-   7. Fill the input as instructed on the screen. Click **Save Changes**.
+#### Requirements
+Some of the requirements to continue with the integration process, are listed below.
+   * WHMCS v5.3.12 - v7.x or later (Tested up to WHMCS v7.6 - running well)
```
Comment (rizdaprasetya) [inline en/snap/with-plugins.md:619]: > (Tested up to WHMCS v7.6 - running well)

This also inconsistencies that we have, that you can fix, the sentence for `Tested up to` is not consistent with other instruction

Comment (rashmikampurath) [reply en/snap/with-plugins.md:619]: I will incorporate these changes.

File: en/snap/with-plugins.md (hunk 6 above)
Comment (rashmikampurath) [reply en/snap/with-plugins.md:507]: I will include the links wherever required.

File: en/snap/with-plugins.md (hunk 5 above)
Comment (rashmikampurath) [reply en/snap/with-plugins.md:116]: As a style, every small step that the user has to perform is required to be listed into separate steps to avoid any confusion. If you think this is not helping, I can change it back to how it was before.

File: en/snap/with-plugins.md (hunk 1 above)
Comment (rashmikampurath) [reply en/snap/with-plugins.md:135]: I will remove the images

File: en/snap/with-plugins.md (hunk 5 above)
Comment (rizdaprasetya) [reply en/snap/with-plugins.md:116]: Yes, let's revert the change of sub-steps converted to numbered steps. Because numbered steps does make the steps seems longer.

Comment (rashmikampurath) [reply en/snap/with-plugins.md:116]: Sure. I will do them and commit the changes today.

Comment (rashmikampurath) [reply en/snap/with-plugins.md:116]: I have made the changes in other sections. Since in the original document, this heading has only one 1 item in the list, I have kept the numbered list.

File: en/snap/with-plugins.md [hunk 8]
```diff
@@ -56,269 +62,308 @@ Here is the list of Content Management System plugins and extensions that are su
...
-    - Fill **Merchant ID, Client Key, and Server key**. You can find [this credential on Midtrans MAP Dashboard](/en/snap/preparation.md).
-    - Other configuration are optional, you may leave it as is.
-
-![WooCommerece Install](./../../asset/image/WooCommerce-install.gif)
 
-#### WooCommerce Plugin Notification Configuration
-Login to your [Midtrans Account](https://account.midtrans.com/login), select your environment (sandbox/production), go to menu `settings` **->** `configuration` and `settings` **->** `Snap Preference` **->** `System Settings`
+   4. On WordPress administration panel, click **Install and activate** the plugin from plugins menu.
+
+The plugin is installed successfully. Proceed to [WooCommerce Plugin Configuration](#wooCommerce-plugin-configuration).
```
Comment (rizdaprasetya) [inline en/snap/with-plugins.md:115]: @rashmikampurath Hey you seem to forget all the basic feedbacks all over again.
```
**Simple Installation**
```
Why does a header got changed to unclickable Bold formatting?

Why is there spacing between numbere lists? Look at your submission render:
> Screen Shot 2020-12-02 at 11 50 10

Compared to original render:
> Screen Shot 2020-12-02 at 11 49 49

Even the original render is much easier to read visually, it grouped and spaced properly. So you were not introducing improvement here.

File: en/snap/with-plugins.md [hunk 9]
```diff
@@ -420,226 +494,284 @@ Customer receives email notification| Receive notification|  Check MAP menu sett
...
+  * Enter **Client key**.
+
+    Other fields are optional. You may leave it as is.
+
+9. Click **Save**.
+   A confirmation message is displayed.
+
+?>***Note***: You can retrieve *Merchant ID, Server key,* and *Client key* on Midtrans MAP Dashboard.<br>
+
+   ![Drupal 8 4](./../../asset/image/drupal8_4.png)
```
Comment (rizdaprasetya) [inline en/snap/with-plugins.md:720]: Ruined the spacing on numbered list again? also why is there two number `9.` ?

File: en/snap/with-plugins.md [hunk 10]
```diff
@@ -350,68 +399,93 @@ Customer receives email notification| Receive notification|  Check MAP menu sett
...
+    * Enter **Merchant ID**.
+    * Enter **Client key.**
+    * Enter **Server key**.
+    * Select desired order status for successful payments, from **Map payment SUCCESS status to** list.
+    * Select desired order status for payment failure, from **Map payment FAILURE status to** list.
+    * Select desired order status for challenge payment, from **Map payment PENDING/CHALLENGE status to** list.
+
+     ?>***Note***: Other fields are optional. You may leave it as is.
+
+The plugin is installed and configured successfully.
```
Comment (rizdaprasetya) [inline en/snap/with-plugins.md:434]: Mind the spacing please

File: en/snap/preparation.md [hunk 11]
```diff
@@ -19,5 +19,6 @@ Sign up for an Account to get your Sandbox API keys ready to test integration.
 Retrieve Sandbox mode API keys that will be used for next sections.
 </div>
 
-?>**Note:**
-Make sure to complete this preparation section to retrieve Client Key & Server Key, before proceeding.
+?>***Note***
+Complete the preparation section to retrieve *Client Key* and *Server Key*, before proceeding.
```
Comment (rizdaprasetya) [inline en/snap/preparation.md:23]: Why did you remove the `:` after `Note` ? it is not even consistent with your other note styling.

File: en/snap/with-plugins.md [hunk 12]
```diff
@@ -56,269 +62,308 @@ Here is the list of Content Management System plugins and extensions that are su
...
+The plugin is installed successfully. Proceed to [WooCommerce Plugin Configuration](#wooCommerce-plugin-configuration).
+
+If you are unable to install, proceed to Manual Installation.
+
+**Manual Installation**
+
+To install Midtrans-WooCommerce plugin manually, follow the steps given below.
+   1. Download the plugin file from the link given above.
 
-#### B. Manual Installation
```
Comment (rizdaprasetya) [inline en/snap/with-plugins.md:90]: In the file `snap/overview.md` you are leaving the `A.` `B.` in the list that is meant to be choice. Why do you remove it here? which consistent approach you want to take?

**Comment by rizdaprasetya** [comment]
@rashmikampurath hey, what is this Pull Request try to submit? a single file like the title says `Snap/with-plugins.md` or many other files? 

Because the PR contains many other files which is very confusing, already discussed multiple times such as here:
- https://github.com/Midtrans/technical-documentation-site/pull/33#issuecomment-735727377
- https://github.com/Midtrans/technical-documentation-site/pull/27#issuecomment-724466380

I'm too tired to explain it all over again TBH, let me just solve it myself. But you need to confirm, is this a single file submission of `Snap/with-plugins.md`?

File: en/snap/with-plugins.md [hunk 13]
```diff
@@ -56,269 +62,308 @@ Here is the list of Content Management System plugins and extensions that are su
...
 
-> **Note:**
->
-> Please make sure to input **http://** or **https://** when filling Notification URL and Redirect URL, according to your web-server configuration. 
-> 
-> If you are not sure, try opening your web URL in a browser, and check the URL is **http** or **https** on the address bar.
-
-2. Go to menu **Settings > Snap Preference > System Settings**
-    * Insert `[your-site-url]/snap/index/finish` link as Finish/Unfinish/Error Redirect URL.
+?> ***Note***: Based on your web-server configuration, please make sure to prefix the *Notification URL* and *Redirect URL*, with **http://** or **https://**.<br>If you are not sure, open your web URL in a browser, and check the URL is **http** or **https** on the address bar.
```
Comment (rizdaprasetya) [inline en/snap/with-plugins.md:340]: While **you change** the appearance of the note,
> Screen Shot 2020-12-02 at 14 20 00

Madhura **does not change** it:
> Screen Shot 2020-12-02 at 14 15 46

Can we have consistency please? Decide which one looks better, and be consistent.

File: en/snap/with-plugins.md (hunk 10 above)
Comment (rashmikampurath) [reply en/snap/with-plugins.md:434]: The spacing between 5. and 6. are deliberate. So that the line "The plugin is installed..." appears as a separate line. The spacing above and below the "Notes" is also deliberate. Otherwise, it is not rendered properly on localhost.

**Comment by rizdaprasetya** [comment]
Resolved & merged in https://github.com/Midtrans/technical-documentation-site/pull/37 closing this duplicate.
